application/frontend/controllers/MfaController.php

```diff
@@ -185,6 +226,46 @@ public function actionVerify($mfaId)
         return $mfa;
     }
 
+    /**
+     * @param $mfaId
+     * @return array|bool
+     * @throws HttpException
+     */
+    public function actionVerifyregistration($mfaId)
+    {
+        $messages = [
+            400 => \Yii::t('app', 'Mfa.InvalidCode'),
+            404 => \Yii::t('app', 'Mfa.VerifyFailure'),
+            429 => \Yii::t('app', 'Mfa.RateLimitFailure'),
+        ];
+
+        $value = self::getVerifyValue();
```

Comment:
Also related to my comment above about returning `null` from the `getVerifyValue()` method:
```suggestion
        $value = self::getVerifyValue();
        if ($value === null) {
            throw new BadRequestHttpException(\Yii::t('app', 'Mfa.MissingValue'));
        }
```